flows/contact.go

```diff
@@ -11,17 +11,32 @@ import (
 
 // Contact represents a single contact
 type Contact struct {
-	uuid     ContactUUID
-	name     string
-	language utils.Language
-	timezone *time.Location
-	urns     URNList
-	groups   GroupList
-	fields   *Fields
+	uuid              ContactUUID
+	name              string
+	language          utils.Language
+	backdownLanguages utils.LanguageList
+	timezone          *time.Location
+	urns              URNList
+	groups            GroupList
+	fields            *Fields
 }
 
+// SetLanguage sets the language for this contact
 func (c *Contact) SetLanguage(lang utils.Language) { c.language = lang }
-func (c *Contact) Language() utils.Language        { return c.language }
+
+// Language gets the language for this contact
+func (c *Contact) Language() utils.Language { return c.language }
+
+// PreferredLanguages gets all languages for this contact in order of preference
+func (c *Contact) PreferredLanguages() utils.LanguageList {
+	var languages utils.LanguageList
+
+	if c.language != utils.NilLanguage {
+		languages = append(languages, c.language)
+	}
+
+	return append(languages, c.backdownLanguages...)
+}
```

Comment:
I think you are right about all the above.

But maybe the reason we are getting confused here is that `backdownLanguages` really isn't an attribute of the contact, but rather an attribute of the environment (or perhaps even the flow).

Maybe these attributes really belong in some kind of environment that is  passed to the flow? (and maybe that's where channels live too?)